src/main/java/ro/utcn/sd/alexh/assignment1/persistence/jdbc/JdbcQuestionVoteRepository.java

```diff
@@ -0,0 +1,81 @@
+package ro.utcn.sd.alexh.assignment1.persistence.jdbc;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+import ro.utcn.sd.alexh.assignment1.entity.QuestionVote;
+import ro.utcn.sd.alexh.assignment1.persistence.api.QuestionVoteRepository;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@RequiredArgsConstructor
+public class JdbcQuestionVoteRepository implements QuestionVoteRepository {
+
+    private final JdbcTemplate template;
+
+    @Override
+    public QuestionVote save(QuestionVote questionVote) {
+        if (questionVote.getQuestionVoteId() != null || findByQuestionAndUser(questionVote.getQuestionId(), questionVote.getUserId()).isPresent()) {
+            update(questionVote);
+        } else {
+            Integer id = insert(questionVote);
+            questionVote.setQuestionVoteId(id);
+        }
+        return questionVote;
+    }
+
+    @Override
+    public Optional<QuestionVote> findById(int id) {
+        List<QuestionVote> questionVotes = template.query("SELECT * FROM question_vote WHERE question_vote_id = ?",
+                (resultSet, i) -> new QuestionVote(
```

Comment:
Again, instead of repeating these mapping lambda's, you can extract them into separate classes and just instantiate these classes here.

Reply:
I was too lazy to modify it, but I'll keep that in mind for future projects. Thank you!